src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/RazorSemanticTokensLegendService.Types.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic;
+
+internal sealed partial class RazorSemanticTokensLegendService
+{
+    public class Types
+    {
+        private static readonly string s_markupAttributeQuoteType = "markupAttributeQuote";
```

Comment:
>private static readonly string s_markupAttributeQuoteType [](http://example.com/codeflow?start=8&length=57)

Is there value in keeping these as separate private static string fields and reading them via reflection as opposed to using a single static string array with all the names? Doesn't seem like we access these individually.

Reply:
We access them individually to read from the dictionary for each public property. Making this one big array would mean reading from the map via array indexing, and I feel like it would be much harder to make sure the data is correct.

Unless I'm missing something about your suggestion?

Reply:
You are right. I guess I'm wondering if maybe they should not be strings but enum names instead and then getter properties would return enum values then? The actual string value doesn't seem to get used other than then key in the array which just maps it to a number which is essentially just the sequence number of that classification?

Reply:
It is a sequence number, yes, but importantly the sequence starts after the last Roslyn token, which we can't know at compile time. If we used an enum, we'd be able to do `nameof` instead of just having the string, but we couldn't rely on the numeric value of the enum being correct.

Reply:
Fair enough :) Missed that part.